### DISCUSSION TITLE
Add Audit Logging System to Worker Tasks

### DISCUSSION THREAD

**Discussion by khareyash05** [langfuse/langfuse#1277] [Ideas]
### Describe the feature or potential improvement

In [https://github.com/langfuse/langfuse/blob/main/web/src/server/api/routers/models.ts](https://github.com/langfuse/langfuse/blob/main/web/src/server/api/routers/models.ts) , we are using the audit logging functionality. I didnt knew about Audit Logging so i went through the link [https://www.datadoghq.com/knowledge-center/audit-logging/](https://www.datadoghq.com/knowledge-center/audit-logging/) . I understood that this is a non-critical task and shouldn't block the other critical tasks. For example - creating/deleting a model. 

We can push this task into the worker directory where we can actually leverage the functionality of Message Queue Systems to schedule these process for later. 
This will also enable easy scaling.

**Comment by khareyash05** [comment]
For reference [https://github.com/Infisical/infisical/blob/main/backend/src/queue/queue-service.ts](https://github.com/Infisical/infisical/blob/main/backend/src/queue/queue-service.ts)

**Comment by marcklingen** [comment]
Hi @khareyash05, well spotted, with the next major release (most likely) most functionality of the ingestion routes will move to the worker behind a redis queue

**Comment by khareyash05** [comment]
> Hi @khareyash05, well spotted, with the next major release (most likely) most functionality of the ingestion routes will move to the worker behind a redis queue

Great! I am willing to help! Let me know if I can help with the migration. I have been recently working around these functionalities using BullMQ, but Redis will not be a major task. Thanks

**Reply by marcklingen** [reply]
Awesome, @maxdeichmann will do a first migration here as it is core infrastructure that affects all teams self-hosting and langfuse cloud. afterwards there are for sure lot's of things to contribute to the new stack
